src/Operator/IsEmptyOperator.php

```diff
@@ -2,22 +2,16 @@
 
 namespace Rx\Operator;
 
+use Rx\DisposableInterface;
 use Rx\ObservableInterface;
-use Rx\Observer\CallbackObserver;
 use Rx\ObserverInterface;
-use Rx\SchedulerInterface;
 
-class IsEmptyOperator implements OperatorInterface
+final class IsEmptyOperator implements OperatorInterface
 {
-    /**
-     * @param \Rx\ObservableInterface $observable
-     * @param \Rx\ObserverInterface $observer
-     * @param \Rx\SchedulerInterface $scheduler
-     * @return \Rx\DisposableInterface
-     */
-    public function __invoke(ObservableInterface $observable, ObserverInterface $observer, SchedulerInterface $scheduler = null)
+
+    public function __invoke(ObservableInterface $observable, ObserverInterface $observer): DisposableInterface
     {
-        $cbObserver = new CallbackObserver(
+        return $observable->subscribe(
```

Comment:
I removed the explicit `CallbackObserver` because it'll get wrapped with `CallbackObserver` anyway.